react/features/base/avatar/components/native/Avatar.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import React from 'react';
+import { Image, Text, View } from 'react-native';
+
+import { connect } from '../../../redux';
+import { type StyleType } from '../../../styles';
+
+import AbstractAvatar, {
+    _mapStateToProps,
+    type Props as AbstractProps,
+    DEFAULT_SIZE
+} from '../AbstractAvatar';
+
+import RemoteAvatar from './RemoteAvatar';
+import styles from './styles';
+
+type Props = AbstractProps & {
+
+    /**
+     * External style of the component.
+     */
+    style?: StyleType
+}
+
+export const DEFAULT_AVATAR = require('../../../../../../images/avatar.png');
+
+/**
+ * Implements an avatar component that has 4 ways to render an avatar:
+ *
+ * - Based on an explicit avatar URI, if provided
+ * - Gravatar, if there is any
+ * - Based on initials generated from name or email
+ * - Default avatar icon, if any of the above fails
+ */
+class Avatar extends AbstractAvatar<Props> {
+
+    _onAvatarLoadError: () => void;
+
+    _onGravatarLoadError: () => void;
+
+    /**
+     * Implements {@code AbstractAvatar#_renderDefaultAvatar}.
+     *
+     * @inheritdoc
+     */
+    _renderDefaultAvatar() {
+        return this._wrapAvatar(
+            <Image
```

Comment:
Why not FastImage?

Reply:
I answered it in another thread. In short, we now prefetch the urls with this component, so it would be a duplication do render it with another